website/content/tutorials/3-constraints-expect/_index.md

```diff
@@ -0,0 +1,1511 @@
+              </define-field>
+            </model>
+          </define-assembly>
+        </model>
+        <constraint>
+          <expect id="memory-same-byte-size" level="WARNING" target="." test="(sum(./memory/byte-size) mod ./memory/byte-size[1]) = 0">
+            <message>All memory modules SHOULD be the same byte size for a computer.</message>
+          </expect>          
+        </constraint>
+      </define-assembly>
+    </model>
+  </define-assembly>
+</METASCHEMA>
+```
+
+The three new `expect` constraints define one mandatory requirement and two optional best practice requirements without exhaustively defining every possible value. Each constraint has a unique `id` for us tool developers to identify the constraints during tool development and potentially use as inputs and outputs for Metaschema processor software itself. These constraints use the flexibility of Metapath to use different context focuses to make the `test` attribute more concise or intuitive for fellow developers.
+
+The `level` attribute defines the impact and potential processing behavior for Metaschema processors during the analysis of document instances. The `level` of a constraint can be `INFORMATIONAL`, `WARNING`, `ERROR`, or `CRITICAL`. Given the new requirements from the stakeholder, failing the `memory-divisible-two` constraint indicates invalid data and necessarily an error condition, so it is an `ERROR`. The `memory-divisible-megabyte` and `memory-same-byte-size` do not encode mandatory requirements, but rather best practices. Therefore, their `level` is `WARNING`.
+
+Our Metaschema processors must evaluate the `test` Metapath against the current context focus from evaluating the `target` (using the `.` operator in Metapath). That result becomes the current context focus in the `test` (the  value of `.` in this expression), and the processor must evaluate the Metapath expression and verify the result is boolean true or false.
+
+The table below identifies the constraints and evaluations of the new constraints in the updated model against the previous sample document instances. Because we are able to control the `target` and `test` context relative to their definitions in the model's assemblies, fields, and flags, we process each result individually for the first two constraints, but process the sequence of all matches for the last constraint all at once. For the latter, we use the more advanced Metapath syntax in the `test` to query each matching assembly of the sequence, filter to only the specific fields' value , and sum them. The data in the table is truncated, but the processor will use process the full `memory` assemblies as the resulting match, and subsequently the `test` Metapath filters on the values of the `byte-size` of each dynamically to compute the sum, then perform the necessary modulo division by the first result (there must always be at least one, never zero, per our module `min-occurs="1"`).
```

Comment:
```suggestion
The table below identifies the constraints and resulting evaluations based on the updated model using the previous sample document instances. Because we are able to control the `target` and `test` context relative to their definitions in the model's assemblies, fields, and flags, we process each result individually for the first two constraints, but process the sequence of all matches for the last constraint all at once. For the latter, we use more advanced Metapath syntax in the `test` to query each matching assembly of the sequence, filter to only the specific fields' value , and sum them. The data in the table is truncated, but the processor will process all the `memory` assemblies as the resulting match, and subsequently the `test` Metapath filters on the values of the `byte-size` of each dynamically to compute the sum, then perform the necessary modulo division by the first result (there must always be at least one, never zero, per our module `min-occurs="1"`).
```

The last sentences are a bit awkward. Consider revising them.

Reply:
Don't I know it. I will think about this then and clean them up instead of just accepting this recommendation. (I am presuming this is what you mean, and I can totally accept that criticism.)

Reply:
Well maybe I will just pull in the batch and then word smith it tomorrow morning, but either way I agree it is awkward.